frontend/amundsen_application/static/js/pages/TableDetailPage/ApplicationDropdown/index.tsx

```diff
@@ -0,0 +1,139 @@
+import DatabricksMenu from './DatabricksMenu';
+import GenericMenu from './GenericMenu';
+import {
+  APPLICATIONS_LABEL,
+  AIRFLOW,
+  DATABRICKS,
+  PRODUCING,
+  CONSUMING,
+} from './constants';
+import './styles.scss';
+
+export interface ApplicationDropdownProps {
+  tableApps: TableApp[];
+}
+
+const getImagePath = (tableAppName) => {
+  switch (tableAppName) {
+    case AIRFLOW.toLowerCase():
+      return '/static/images/airflow.jpeg';
+    case DATABRICKS.toLowerCase():
+      return '/static/images/icons/logo-databricks.png';
+    default:
+      return '/static/images/icons/application.svg';
+  }
+};
+
+const isAirflowOrDatabricksApp = (appName) =>
+  appName.toLowerCase() === AIRFLOW.toLowerCase() ||
+  appName.toLowerCase() === DATABRICKS.toLowerCase();
+
+const sortByNameOrId = (a, b) =>
+  a.name.localeCompare(b.name) || a.id.localeCompare(b.id);
+
+const hasSameNameAndKind = (app, name, kind) =>
+  // Checks if the app matches the given name and kind
+  // If its kind is empty, check if the given kind is the default Producing value
+  app.name === name &&
+  ((app.kind && app.kind === kind) || (!app.kind && kind === PRODUCING));
+
+const getSortedAppKinds = (apps: TableApp[]) => {
+  // Sort app kinds by Producing then Consuming if they exist in the list, and then all others following
+  // If no kind is specified, default to Producing
+  const appKinds: string[] = [
+    ...new Set(apps.map((app) => (app.kind ? app.kind : PRODUCING))),
+  ];
+
+  const producingKind = appKinds.filter(
+    (kind) => kind.toLowerCase() === PRODUCING
+  );
+  const consumingKind = appKinds.filter(
+    (kind) => kind.toLowerCase() === CONSUMING
+  );
+  const remainingKinds = appKinds.filter(
+    (kind) =>
+      kind.toLowerCase() !== PRODUCING && kind.toLowerCase() !== CONSUMING
+  );
+
+  return [...producingKind, ...consumingKind, ...remainingKinds];
+};
+
+const handleClick = (event) => {
+  event.stopPropagation();
+  logClick(event);
+};
+
+const getDropdownMenuContents = (tableApps) => {
+  if (tableApps[0].name.toLowerCase() === AIRFLOW.toLowerCase()) {
+    return (
+      <AirflowMenu
+        tableApps={tableApps}
+        getSortedAppKinds={getSortedAppKinds}
+        hasSameNameAndKind={hasSameNameAndKind}
+        handleClick={handleClick}
+      />
+    );
+  }
+  if (tableApps[0].name.toLowerCase() === DATABRICKS.toLowerCase()) {
+    return (
+      <DatabricksMenu
+        tableApps={tableApps}
+        getSortedAppKinds={getSortedAppKinds}
+        hasSameNameAndKind={hasSameNameAndKind}
+        handleClick={handleClick}
+      />
+    );
+  }
+  return (
+    <GenericMenu
+      tableApps={tableApps}
+      getSortedAppKinds={getSortedAppKinds}
+      hasSameNameAndKind={hasSameNameAndKind}
+      handleClick={handleClick}
+    />
+  );
+};
+
+const ApplicationDropdown: React.FC<ApplicationDropdownProps> = ({
+  tableApps,
+}: ApplicationDropdownProps) => {
+  if (tableApps === null || tableApps.length === 0) return null;
```

Comment:
We don't do one liners like these, is an historic pattern we follow:
```suggestion
  if (tableApps === null || tableApps.length === 0) {
      return null;
  }
```